infra/ansible/roles/build_srcs/tasks/main.yaml

```diff
@@ -8,8 +8,14 @@
   # ansible.builtin.git wants to fetch the entire history, so check out the pin manually
   ansible.builtin.shell:
     cmd: |
-      git fetch origin {{ torch_pin.stdout }}
-      git checkout --recurse-submodules {{ torch_pin.stdout }}
+      if [[ {{ torch_pin.stdout }} = \#* ]]; then
+        PRNUM="${torch_pin.stdout//[!0-9]/}"
+        git fetch origin "pull/$PRNUM/head:$PRNUM"
```

Comment:
You can simplify this a bit:

```
  ...
  git fetch origin "pull/$PRNUM/head
else
  git fetch origin {{ torch_pin.stdout }}
fi
git checkout --recurse-submodules FETCH_HEAD
```

Either ref you fetch will be tagged `FETCH_HEAD` immediately after, so there's no need to create a new branch called `$PRNUM`